src/socket.rs

```diff
@@ -1017,6 +1119,28 @@ impl Drop for Socket {
     }
 }
 
+/// Fix-up a created socket by setting the proper portability flags.
+#[allow(unused)]
+fn fixup_socket(mut sock: Socket) -> io::Result<Socket> {
+    #[cfg(windows)]
+    {
```

Comment:
Move this to `accept`, not needed in `Socket::new` or `Socket::new_pair`.

Reply:
It is not called in `Socket::new()` on windows, and `new_pair()` doesn't exist on Windows. Doing it here keeps all the platform specific fix-ups in one place. But sure, I'll move it.